RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc

```diff
@@ -257,12 +272,20 @@ std::vector<CSCSegment> CSCSegAlgoRU::buildSegments(const CSCChamber* aChamber,
       search_disp = false;
       aState.doCollisions = true;
       aState.dRMax = 2.0;
-      aState.dPhiMax = 0.25*aState.dPhiMax;
-      aState.dRIntMax = 0.25*aState.dRIntMax;
-      aState.dPhiIntMax = 0.25*aState.dPhiIntMax;
-      aState.chi2Norm_2D_ = 0.1*aState.chi2Norm_2D_;
       aState.chi2_str_ = 100;
-      aState.chi2Max = 0.25*aState.chi2Max;
+      if(enlarge){	   
+        aState.dPhiMax = 0.25*aState.dPhiMax;
+        aState.dRIntMax = 0.25*aState.dRIntMax;
+        aState.dPhiIntMax = 0.25*aState.dPhiIntMax;
+        aState.chi2Norm_2D_ = 0.1*aState.chi2Norm_2D_;
+        aState.chi2Max = 0.25*aState.chi2Max;
+      } else {
+        aState.dPhiMax = 0.5*aState.dPhiMax;
+        aState.dRIntMax = 0.5*aState.dRIntMax;
```

Comment:
please do not copy-paste essentially the same lines of code.
Define a scale constant (1. for enlarge=false; and 2. for enlarge = true)